lib/puppet/provider/githubreleases_download/githubreleases_download.rb

```diff
@@ -6,8 +6,19 @@
 
   def create
     download_url = get_download_url(@resource[:author], @resource[:repository], @resource)
-    download = open(download_url)
-    IO.copy_stream(download, @resource[:target])
+    uri = URI.parse(download_url)
+
+    # Follow redirects
+    response = Net::HTTP.get_response(uri)
+    while response.code.to_i == 302  # Redirect status code
```

Comment:
Maybe add 301 as well here.

Reply:
Thanks for the quick review! Changed the check to check both codes.